src/NUnitFramework/framework/Constraints/AnyOfConstraint.cs

```diff
@@ -0,0 +1,177 @@
+// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// <see cref="AnyOfConstraint"/> is used to determine whether item is equal to any of parameters
+    /// </summary>
+    public class AnyOfConstraint : Constraint
+    {
+        private readonly IEnumerable _expected;
+
+        /// <summary>
+        /// The NUnitEqualityComparer in use for this constraint
+        /// </summary>
+        private readonly NUnitEqualityComparer _comparer = NUnitEqualityComparer.Default;
+
+        /// <summary>
+        /// Construct a <see cref="AnyOfConstraint"/>
+        /// </summary>
+        /// <param name="expected">Expected collection</param>
+        public AnyOfConstraint(IEnumerable expected) : base(expected)
+        {
+            _expected = expected;
+        }
+
+        /// <summary>
+        /// The Description of what this constraint tests, for
+        /// use in messages and in the ConstraintResult.
+        /// </summary>
+        public override string Description
+        {
+            get
+            {
+                return "any of " + MsgUtils.FormatValue(_expected);
+            }
+        }
+
+        /// <summary>
+        /// Returns the constraint DisplayName followed by arguments within angle brackets.
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            var expectedStrings = _expected.Cast<object>().Select(MsgUtils.FormatValue).ToArray();
+            return $"<{DisplayName.ToLower()} {string.Join(", ", expectedStrings)}>";
```

Comment:
While it will do no real harm to have the commas, this is different from the formatting of any other constraint. See `RangeConstrant` for example. I'd prefer to have them consistent.